### PR TITLE
Update changelog for 2.15.0-rc.1

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This updates the changelog. In the future these "changelog update PRs" will be partt of the version they are describing (i.e. checking out 2.15.0 will include the changelog for 2.15.0 itself), with this I just wanted to learn how to do it. I had to hack our changelog generator a bit and that's why I didn't use the `changelog-gen.sh` this time.

As this is the log for a new minor release, I tried to group multiple changes for bigger features together and roughly eye-ball what a "highlight" should be.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by xrstf** [comment]
/hold

needs approval from CL and UI teams

**Review by xmudrii — APPROVED** [review]
/lgtm

**Review by xmudrii — APPROVED** [review]
/lgtm

File: CHANGELOG.md
Comment (scheeles) [inline CHANGELOG.md:21]: was this not available with 2.14 or should we write `Enhance...`?

**Review by scheeles — APPROVED** [review]

**Comment by maciaszczykm** [comment]
LGTM from UI side

@zreigz Can you do the final approval?

File: CHANGELOG.md
Comment (xrstf) [reply CHANGELOG.md:21]: It might have been backported to a 2.14 patch release, but it should not have been in 2.14.0.

**Pull request merged**
